Sources/TuistKit/Services/Registry/RegistryLoginService.swift

```diff
@@ -67,31 +79,79 @@ struct RegistryLoginService {
 
         ServiceContext.current?.logger?.info("Logging into the registry...")
         let serverURL = try serverURLService.url(configServerURL: config.url)
+        let registryURL = serverURL.appending(path: "api/accounts/\(accountHandle)/registry/swift")
 
-        let token: String
         if ciChecker.isCI() {
-            switch try await serverAuthenticationController.authenticationToken(serverURL: serverURL) {
-            case let .project(projectToken):
-                token = projectToken
-            case .user, .none:
-                throw RegistryLoginServiceError.missingProjectToken
-            }
+            try await registryCILogin(
+                registryURL: registryURL,
+                serverURL: serverURL,
+                path: path
+            )
         } else {
-            token = try await createAccountTokenService.createAccountToken(
+            try await registryUserLogin(
                 accountHandle: accountHandle,
-                scopes: [.accountRegistryRead],
+                registryURL: registryURL,
                 serverURL: serverURL
             )
         }
 
-        let registryURL = serverURL.appending(path: "api/accounts/\(accountHandle)/registry/swift")
+        ServiceContext.current?.logger?.info("Successfully logged in to the \(accountHandle) registry 🎉")
+    }
 
+    private func registryCILogin(
+        registryURL: URL,
+        serverURL: URL,
+        path: AbsolutePath
+    ) async throws {
+        switch try await serverAuthenticationController.authenticationToken(serverURL: serverURL) {
+        case let .project(projectToken):
+            let manifest = try await manifestFilesLocator.locatePackageManifest(at: path)
+            if try await manifestFilesLocator.locatePackageManifest(at: path) == nil {
```

Comment:
Is there a reason we're duplicating the call here? I'd assume that this would work:
```suggestion
            let manifest = try await manifestFilesLocator.locatePackageManifest(at: path)
            if manifest == nil {
```

Reply:
Nope, I added the extra one for some debugging and it's no longer needed. Thanks!